app/cocina_builders/contributor_affiliation_cocina_builder.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Generates contributor affiliation for the Cocina object
+class ContributorAffiliationCocinaBuilder
+  def self.call(...)
+    new(...).call
+  end
+
+  def initialize(department:, institution:, uri:)
+    @department = department
+    @institution = institution
+    @uri = uri
+  end
+
+  attr_reader :department, :institution, :uri
+
+  def call
+    {}.tap do |affiliation|
+      affiliation[:type] = 'affiliation'
+      affiliation[:structuredValue] = [
+        generate_descriptive_value,
+        { value: department }
+      ].compact_blank
```

Comment:
What could be blank or nill?

Reply:
I updated this a bit, but `compact` is still necessary to remove `nil` from department.